src/miner.cpp

```diff
@@ -312,6 +353,13 @@ void BlockAssembler::addPackageTxs(int &nPackagesSelected, int &nDescendantsUpda
     // mempool has a lot of entries.
     const int64_t MAX_CONSECUTIVE_FAILURES = 1000;
     int64_t nConsecutiveFailed = 0;
+    uint64_t nPaidTxSize = 0;
+    auto freeTxMaxSizeInBlock = chainparams.GetConsensus().freeTxMaxSizeInBlock;
+
+    if (freeTxMaxSizeInBlock > (nBlockMaxWeight * .25) ) {
```

Comment:
No sense in validating the consensus parameter here. It should be taken as it is.